Core/include/Acts/Geometry/MaterialDesignatorBlueprintNode.hpp

```diff
@@ -0,0 +1,93 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/ProtoBinning.hpp"
+#include "Acts/Geometry/BlueprintNode.hpp"
+#include "Acts/Geometry/PortalShell.hpp"
+
+#include <variant>
+
+namespace Acts {
+
+/// This node type registers material proxies into its child volume during the
+/// blueprint construction. It is configured ahead of time which volume faces to
+/// mark up, and how do to so.
+/// @note This node can only have a single child. This is not an error during
+///       tree building, but during geometry construction.
+/// @note This currently only supports a cylinder volume child
+class MaterialDesignatorBlueprintNode final : public BlueprintNode {
+ public:
+  // @TODO: This needs cuboid volume storage as well
+  // @TODO: I don't love the type
+  using BinningConfig = std::variant<std::vector<
+      std::tuple<CylinderVolumeBounds::Face, Experimental::ProtoBinning,
+                 Experimental::ProtoBinning>>>;
```

Comment:
what does the `variant` do?

Reply:
- I want to be able to configure the same material designator node for different child volumes (currently only cylinder, cuboid in the future)
- I want to move away from the catch-all `BoundarySurfaceFace` enum towards typesafe enums per volume bounds type
- Given the two above, I need to define a config type that can store different tuples depending on what the expected child type is going to be. I thought `variant` makes the most sense here, but it currently only contains the cylinder face type.
  - Like I say in the `@TODO` comment, I don't love this interface, but I wanted move ahead with this before potentially refactoring it later on.